File: top-interview-questions-medium/Sorting and Searching/merge-intervals/56.py
# Definition for an interval.
# class Interval:
#     def __init__(self, s=0, e=0):
#         self.start = s
#         self.end = e


class Solution:
    def merge(self, intervals):
        """
        :type intervals: List[Interval]
        :rtype: List[Interval]
        """
        if not len(intervals): return []
        intervals.sort(key=lambda ele: ele.start)
        result = [intervals[0]]
        for i in range(1, len(intervals)):
            curr = intervals[i]
            last = result[len(result) - 1]
            if curr.start > last.end:
                result.append(curr)
            elif curr.end >= last.end:
                last.end = curr.end

        return result

File: top-interview-questions-medium/Sorting and Searching/merge-intervals/56.js
/**
 * Definition for an interval.
 * function Interval(start, end) {
 *     this.start = start;
 *     this.end = end;
 * }
 */
/**
 * @param {Interval[]} intervals
 * @return {Interval[]}
 */
var merge = function(intervals) {
  if (!intervals.length) return []
  intervals.sort((a, b) => a.start - b.start)

  const result = [intervals[0]]
  for (let i = 1; i < intervals.length; i++) {
    let curr = intervals[i]
    let last = result[result.length - 1]
    if (curr.start > last.end) {
      result.push(curr)
    } else if (curr.end >= last.end) {
      last.end = curr.end
    }
  }
  return result
}
